# Neanderthal code

### Requirements:
Python 3

Python modules:
* pandas 1.0.3

---

### Description of code
    neanderthal_gwas.ipynb
This code investigates if Neanderthal-introgressed SNPs (present in Chen introgressed sequences) have been associated with any immune-related diseases, including infectious diseases, allergic diseases, autoimmune diseases and autoinflammatory diseases, using data from the NHGRI-EBI GWAS Catalog.

---

### Datasets
Neanderthal-introgressed SNPs from:
1. Dannemann M, Prufer K & Kelso J. Functional implications of Neandertal introgression in modern humans. *Genome Biol* 2017 **18**:61.
2. Simonti CN *et al.* The phenotypic legacy of admixture between modern humans and Neandertals. *Science* 2016 **351**:737-41.  
&nbsp; 

Neanderthal-introgressed sequences by Chen *et al.* from:
* Chen L *et al.* Identifying and interpreting apparent Neanderthal ancestry in African individuals. *Cell* 2020 **180**:677-687.  
&nbsp; 

MAFs from:
* [1000 genomes](https://www.internationalgenome.org/data/)  
&nbsp;

Genotyped SNPs and monocyte, gene-expression data from:
* Fairfax BP *et al.* Innate immune activity conditions the effect of regulatory variants upon monocyte gene expression. *Science* 2014 **343**:1246949.  
&nbsp;

Recomputed Fairfax *et al.* eQTLs from:
* [EMBL-EBI eQTL Catalogue](https://www.ebi.ac.uk/eqtl/Data_access/)  
&nbsp;

GWAS summary statistics from:
* [GWAS Catalog](https://www.ebi.ac.uk/gwas/docs/file-downloads)
