{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explore overlap between Neanderthal-introgressed SNPs and Nedelec eQTLs\n",
    "\n",
    "This code explores if any Neanderthal-introgressed SNPs are present in the list of macrophage extreme response cis eQTLs identified by Nedelec *et al*. These QTLs comprise condition-specific eQTLs, response eQTLs (reQTLs) that are associated with gene expression levels after infection, and alternative splicing QTLs (asQTLs) that are associated with transcript isoform usage. Macrophages were subjected to 3 experimental conditions: *Listeria*-infected, *Salmonella*-infected and non-infected.\n",
    "\n",
    "Neanderthal SNPs from:\n",
    "1. Dannemann M, Prufer K & Kelso J. Functional implications of Neandertal introgression in modern humans. Genome Biol 2017 18:61.\n",
    "2. Simonti CN *et al*. The phenotypic legacy of admixture between modern humans and Neandertals. Science 2016 351:737-41.\n",
    "\n",
    "Macrophages eQTLs from:\n",
    "1. Nedelec Y *et al*. Genetic Ancestry and Natural Selection Drive Population Differences in Immune Responses to Pathogens. Cell 2016 167:657-669 e21."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import libraries\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load allpop_df containing Neanderthal SNPs\n",
    "allpop_df = pd.read_csv(\"../neanderthal/allpop_df.csv\", usecols=[\"Chromosome\", \"Position\", \"Source\", \"ID\",\n",
    "                                                  \"ESN\", \"YRI\", \"MSL\", \"GWD\", \"LWK\"])\n",
    "allpop_df = allpop_df.dropna()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Condition-specific eQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs10001483</td>\n",
       "      <td>TBCK</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10001849</td>\n",
       "      <td>PLAC8</td>\n",
       "      <td>NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10001970</td>\n",
       "      <td>RPS3A</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000255</td>\n",
       "      <td>ST3GAL4</td>\n",
       "      <td>NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000354</td>\n",
       "      <td>FMNL1</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ID     Gene Set\n",
       "0  rs10001483     TBCK   S\n",
       "1  rs10001849    PLAC8  NI\n",
       "2  rs10001970    RPS3A   S\n",
       "3   rs1000255  ST3GAL4  NI\n",
       "4   rs1000354    FMNL1   S"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load eQTL data\n",
    "ned_eqtl = pd.read_excel(\"nedelec_eQTL.xlsx\", \"A) cis eQTL\", skiprows=[0,1])\n",
    "ned_eqtl.columns = ned_eqtl.iloc[0, :]\n",
    "ned_eqtl.drop(ned_eqtl.index[0], inplace=True)\n",
    "ned_eqtl.rename(columns={\"external_gene_name\": \"Gene\"}, inplace=True)\n",
    "\n",
    "# Separate SNP IDs onto different rows based on treatment\n",
    "ned_eqtl_1 = ned_eqtl.loc[:, [\"Gene\", \"NI_top_snp_id\"]]\n",
    "ned_eqtl_1['Set'] = 'NI'\n",
    "ned_eqtl_1.rename(columns={\"NI_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_eqtl_2 = ned_eqtl.loc[:, [\"Gene\", \"L_top_snp_id\"]]\n",
    "ned_eqtl_2['Set'] = 'L'\n",
    "ned_eqtl_2.rename(columns={\"L_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_eqtl_3 = ned_eqtl.loc[:, [\"Gene\", \"S_top_snp_id\"]]\n",
    "ned_eqtl_3['Set'] = 'S'\n",
    "ned_eqtl_3.rename(columns={\"S_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_eqtl = pd.concat([ned_eqtl_1, ned_eqtl_2, ned_eqtl_3])\n",
    "ned_eqtl = ned_eqtl.sort_values([\"Gene\", \"ID\"], ascending=(True, True))\n",
    "\n",
    "# Combine rows with same Gene and ID\n",
    "ned_eqtl = ned_eqtl.groupby([\"Gene\", \"ID\"]).agg(','.join).reset_index()\n",
    "ned_eqtl = ned_eqtl.groupby([\"ID\"]).agg(','.join).reset_index()\n",
    "\n",
    "ned_eqtl.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of intersecting SNPs: 98\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Source</th>\n",
       "      <th>ID</th>\n",
       "      <th>ESN</th>\n",
       "      <th>GWD</th>\n",
       "      <th>LWK</th>\n",
       "      <th>YRI</th>\n",
       "      <th>MSL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>54310748</td>\n",
       "      <td>both</td>\n",
       "      <td>rs7556609</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>YIPF1</td>\n",
       "      <td>NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>212429194</td>\n",
       "      <td>both</td>\n",
       "      <td>rs17018847</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.006</td>\n",
       "      <td>PPP2R5A</td>\n",
       "      <td>NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>64702053</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs2893913</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.045</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.029</td>\n",
       "      <td>EGR2</td>\n",
       "      <td>NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>9370093</td>\n",
       "      <td>both</td>\n",
       "      <td>rs4910456</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>IPO7</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>21568031</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs12322891</td>\n",
       "      <td>0.045</td>\n",
       "      <td>0.027</td>\n",
       "      <td>0.106</td>\n",
       "      <td>0.019</td>\n",
       "      <td>0.018</td>\n",
       "      <td>GOLT1B</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome   Position        Source          ID    ESN    GWD    LWK  \\\n",
       "0           1   54310748          both   rs7556609  0.000  0.000  0.000   \n",
       "1           1  212429194          both  rs17018847  0.000  0.000  0.000   \n",
       "2          10   64702053  simonti_only   rs2893913  0.020  0.018  0.045   \n",
       "3          11    9370093          both   rs4910456  0.000  0.000  0.000   \n",
       "4          12   21568031  simonti_only  rs12322891  0.045  0.027  0.106   \n",
       "\n",
       "     YRI    MSL     Gene Set  \n",
       "0  0.000  0.000    YIPF1  NI  \n",
       "1  0.000  0.006  PPP2R5A  NI  \n",
       "2  0.023  0.029     EGR2  NI  \n",
       "3  0.000  0.000     IPO7   L  \n",
       "4  0.019  0.018   GOLT1B   L  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Intersecting SNPs in Neanderthal & Nedelec eQTL datasets\n",
    "ned_eqtl_df = allpop_df.merge(ned_eqtl, how='inner', on=['ID'])\n",
    "print(\"Number of intersecting SNPs: \" + str(len(ned_eqtl_df)))\n",
    "ned_eqtl_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ESN</th>\n",
       "      <th>YRI</th>\n",
       "      <th>MSL</th>\n",
       "      <th>GWD</th>\n",
       "      <th>LWK</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>98.000000</td>\n",
       "      <td>98.000000</td>\n",
       "      <td>98.000000</td>\n",
       "      <td>98.000000</td>\n",
       "      <td>98.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>0.057490</td>\n",
       "      <td>0.056551</td>\n",
       "      <td>0.059204</td>\n",
       "      <td>0.060847</td>\n",
       "      <td>0.063112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>0.220314</td>\n",
       "      <td>0.220687</td>\n",
       "      <td>0.219329</td>\n",
       "      <td>0.218350</td>\n",
       "      <td>0.219690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004000</td>\n",
       "      <td>0.005000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>0.005000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012000</td>\n",
       "      <td>0.013000</td>\n",
       "      <td>0.018750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.996000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             ESN        YRI        MSL        GWD        LWK\n",
       "count  98.000000  98.000000  98.000000  98.000000  98.000000\n",
       "mean    0.057490   0.056551   0.059204   0.060847   0.063112\n",
       "std     0.220314   0.220687   0.219329   0.218350   0.219690\n",
       "min     0.000000   0.000000   0.000000   0.000000   0.000000\n",
       "25%     0.000000   0.000000   0.000000   0.000000   0.000000\n",
       "50%     0.000000   0.000000   0.000000   0.004000   0.005000\n",
       "75%     0.005000   0.000000   0.012000   0.013000   0.018750\n",
       "max     1.000000   1.000000   1.000000   0.996000   1.000000"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# MAF for intersecting SNPs in African populations\n",
    "ned_eqtl_df[['ESN', 'YRI', 'MSL', 'GWD', 'LWK']].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1mESN\u001b[0m\n",
      "Number: 18\n",
      "\u001b[1mYRI\u001b[0m\n",
      "Number: 17\n",
      "\u001b[1mMSL\u001b[0m\n",
      "Number: 27\n",
      "\u001b[1mGWD\u001b[0m\n",
      "Number: 26\n",
      "\u001b[1mLWK\u001b[0m\n",
      "Number: 28\n"
     ]
    }
   ],
   "source": [
    "# Number of SNPs with MAF > 0.01 in African populations\n",
    "for pop in ['ESN', 'YRI', 'MSL', 'GWD', 'LWK']:\n",
    "    df = ned_eqtl_df[ned_eqtl_df[pop] > 0.01]\n",
    "    print('\\033[1m' + pop + '\\033[0m', 'Number: ' + str(len(df)), sep='\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1mExpression context of intersecting SNPs\u001b[0m\n",
      "NI    44\n",
      "L     35\n",
      "S     31\n",
      "dtype: int64\n",
      "\u001b[1mAssociated genes\u001b[0m\n",
      "CYB5A        2\n",
      "HPCA         2\n",
      "SLC24A4      2\n",
      "CCT2         2\n",
      "NFKBIA       2\n",
      "            ..\n",
      "TUBB3        1\n",
      "HIST1H2BJ    1\n",
      "NRF1         1\n",
      "SEC31A       1\n",
      "REXO4        1\n",
      "Length: 94, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Expression context of intersecting SNPs\n",
    "set = ned_eqtl_df['Set'].str.split(',', expand=True).stack().value_counts()\n",
    "print('\\033[1mExpression context of intersecting SNPs\\033[0m', set, sep='\\n')\n",
    "\n",
    "genes = ned_eqtl_df['Gene'].str.split(',', expand=True).stack().value_counts()\n",
    "print(\"\\033[1mAssociated genes\\033[0m\", genes, sep='\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### reQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000269</td>\n",
       "      <td>C20orf26</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10003</td>\n",
       "      <td>IMPAD1</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10005213</td>\n",
       "      <td>KLHL2</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs10005401</td>\n",
       "      <td>HNRNPDL</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10008992</td>\n",
       "      <td>C4orf21</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ID      Gene Set\n",
       "0   rs1000269  C20orf26   L\n",
       "1     rs10003    IMPAD1   L\n",
       "2  rs10005213     KLHL2   L\n",
       "3  rs10005401   HNRNPDL   S\n",
       "4  rs10008992   C4orf21   S"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load reQTL data\n",
    "ned_reqtl = pd.read_excel(\"nedelec_eQTL.xlsx\", \"B) cis reQTL\", skiprows=[0,1])\n",
    "ned_reqtl.columns = ned_reqtl.iloc[0, :]\n",
    "ned_reqtl.drop(ned_reqtl.index[0], inplace=True)\n",
    "ned_reqtl.rename(columns={\"external_gene_name\": \"Gene\"}, inplace=True)\n",
    "\n",
    "# Separate SNP IDs onto different rows based on treatment\n",
    "ned_reqtl_1 = ned_reqtl.loc[:, [\"Gene\", \"FC_L_top_snp_id\"]]\n",
    "ned_reqtl_1['Set'] = 'L'\n",
    "ned_reqtl_1.rename(columns={\"FC_L_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_reqtl_2 = ned_reqtl.loc[:, [\"Gene\", \"FC_S_top_snp_id\"]]\n",
    "ned_reqtl_2['Set'] = 'S'\n",
    "ned_reqtl_2.rename(columns={\"FC_S_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_reqtl = pd.concat([ned_reqtl_1, ned_reqtl_2])\n",
    "ned_reqtl = ned_reqtl.sort_values([\"Gene\", \"ID\"], ascending=(True, True))\n",
    "\n",
    "# Combine rows with same Gene and ID\n",
    "ned_reqtl = ned_reqtl.groupby([\"Gene\", \"ID\"]).agg(','.join).reset_index()\n",
    "ned_reqtl = ned_reqtl.groupby([\"ID\"]).agg(','.join).reset_index()\n",
    "\n",
    "ned_reqtl.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of intersecting SNPs: 53\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Source</th>\n",
       "      <th>ID</th>\n",
       "      <th>ESN</th>\n",
       "      <th>GWD</th>\n",
       "      <th>LWK</th>\n",
       "      <th>YRI</th>\n",
       "      <th>MSL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>19892021</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs75061720</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.014</td>\n",
       "      <td>0.012</td>\n",
       "      <td>MINOS1,NBL1</td>\n",
       "      <td>S,L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>36257328</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs61879262</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>COMMD9</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12</td>\n",
       "      <td>54101479</td>\n",
       "      <td>both</td>\n",
       "      <td>rs73104021</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.005</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>CALCOCO1</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>13</td>\n",
       "      <td>73552634</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs75497275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>KLF5</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>21004628</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs2319366</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004</td>\n",
       "      <td>0.005</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>TMEM55B</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome  Position        Source          ID  ESN    GWD    LWK    YRI  \\\n",
       "0           1  19892021  simonti_only  rs75061720  0.0  0.009  0.000  0.014   \n",
       "1          11  36257328  simonti_only  rs61879262  0.0  0.000  0.000  0.000   \n",
       "2          12  54101479          both  rs73104021  0.0  0.000  0.005  0.000   \n",
       "3          13  73552634  simonti_only  rs75497275  0.0  0.004  0.000  0.000   \n",
       "4          14  21004628  simonti_only   rs2319366  0.0  0.004  0.005  0.000   \n",
       "\n",
       "     MSL         Gene  Set  \n",
       "0  0.012  MINOS1,NBL1  S,L  \n",
       "1  0.000       COMMD9    S  \n",
       "2  0.000     CALCOCO1    L  \n",
       "3  0.000         KLF5    S  \n",
       "4  0.000      TMEM55B    L  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Intersecting SNPs in Neanderthal & Nedelec reQTL datasets\n",
    "ned_reqtl_df = allpop_df.merge(ned_reqtl, how='inner', on=['ID'])\n",
    "print(\"Number of intersecting SNPs: \" + str(len(ned_reqtl_df)))\n",
    "ned_reqtl_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1mExpression context of intersecting SNPs\u001b[0m\n",
      "L    32\n",
      "S    24\n",
      "dtype: int64\n",
      "\u001b[1mAssociated genes\u001b[0m\n",
      "CALCOCO1      2\n",
      "E2F6          2\n",
      "TMEM55B       1\n",
      "OXNAD1        1\n",
      "DUSP12        1\n",
      "COMMD9        1\n",
      "ALKBH7        1\n",
      "PRDM1         1\n",
      "MPP6          1\n",
      "KLF5          1\n",
      "CCDC146       1\n",
      "KNSTRN        1\n",
      "ITPR3         1\n",
      "HIST2H2AA3    1\n",
      "ACAP2         1\n",
      "MINOS1        1\n",
      "NR4A1         1\n",
      "SNW1          1\n",
      "PHRF1         1\n",
      "HMGN4         1\n",
      "CCDC42B       1\n",
      "PNISR         1\n",
      "TRAF3IP2      1\n",
      "PPP1R37       1\n",
      "PFDN6         1\n",
      "AC187652.1    1\n",
      "FAM120A       1\n",
      "TRIM68        1\n",
      "ELP4          1\n",
      "IQCD          1\n",
      "PSMD14        1\n",
      "PTPDC1        1\n",
      "ATP6V1C1      1\n",
      "HECA          1\n",
      "RAB11FIP2     1\n",
      "PTPLAD2       1\n",
      "MED11         1\n",
      "ZNF318        1\n",
      "TMEM41B       1\n",
      "MIER3         1\n",
      "MTFR1         1\n",
      "ADCK1         1\n",
      "RAB3IP        1\n",
      "DDX21         1\n",
      "RUVBL2        1\n",
      "ZNF189        1\n",
      "NBL1          1\n",
      "FCGR2C        1\n",
      "CCT2          1\n",
      "FZD6          1\n",
      "DENND5A       1\n",
      "YTHDF1        1\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Expression context of intersecting SNPs\n",
    "set = ned_reqtl_df['Set'].str.split(',', expand=True).stack().value_counts()\n",
    "print('\\033[1mExpression context of intersecting SNPs\\033[0m', set, sep='\\n')\n",
    "\n",
    "genes = ned_reqtl_df['Gene'].str.split(',', expand=True).stack().value_counts()\n",
    "print(\"\\033[1mAssociated genes\\033[0m\", genes, sep='\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### asQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs10000006</td>\n",
       "      <td>CYP2U1,SGMS2</td>\n",
       "      <td>NI,L,S,L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000012</td>\n",
       "      <td>CRIPAK</td>\n",
       "      <td>NI,L,S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000002</td>\n",
       "      <td>MAP6D1</td>\n",
       "      <td>NI,L,S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs10000037</td>\n",
       "      <td>TLR6</td>\n",
       "      <td>NI,L,S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10000130</td>\n",
       "      <td>AC110615.1,CXCL10,CXCL11,CXCL9</td>\n",
       "      <td>NI,L,S,NI,L,S,NI,L,S,NI,L,S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ID                            Gene                          Set\n",
       "0  rs10000006                    CYP2U1,SGMS2                     NI,L,S,L\n",
       "1  rs10000012                          CRIPAK                       NI,L,S\n",
       "2   rs1000002                          MAP6D1                       NI,L,S\n",
       "3  rs10000037                            TLR6                       NI,L,S\n",
       "4  rs10000130  AC110615.1,CXCL10,CXCL11,CXCL9  NI,L,S,NI,L,S,NI,L,S,NI,L,S"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load asQTL data\n",
    "ned_asqtl = pd.read_excel(\"nedelec_eQTL.xlsx\", \"C) cis asQTL\", skiprows=[0, 1])\n",
    "ned_asqtl.columns = ned_asqtl.iloc[0, :]\n",
    "ned_asqtl.drop(ned_asqtl.index[0], inplace=True)\n",
    "ned_asqtl.rename(columns={\"external_gene_name\": \"Gene\"}, inplace=True)\n",
    "\n",
    "# Separate SNP IDs onto different rows based on treatment\n",
    "ned_asqtl_1 = ned_asqtl.loc[:, [\"Gene\", \"NI_top_snp_id\"]]\n",
    "ned_asqtl_1['Set'] = 'NI'\n",
    "ned_asqtl_1.rename(columns={\"NI_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_asqtl_2 = ned_asqtl.loc[:, [\"Gene\", \"L_top_snp_id\"]]\n",
    "ned_asqtl_2['Set'] = 'L'\n",
    "ned_asqtl_2.rename(columns={\"L_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_asqtl_3 = ned_asqtl.loc[:, [\"Gene\", \"S_top_snp_id\"]]\n",
    "ned_asqtl_3['Set'] = 'S'\n",
    "ned_asqtl_3.rename(columns={\"S_top_snp_id\": \"ID\"}, inplace=True)\n",
    "ned_asqtl = pd.concat([ned_asqtl_1, ned_asqtl_2, ned_asqtl_3])\n",
    "\n",
    "# Format dataframe\n",
    "ned_asqtl = ned_asqtl.sort_values([\"Gene\", \"ID\"], ascending=(True, True))\n",
    "ned_asqtl = ned_asqtl.drop_duplicates(subset=['Gene', 'ID', 'Set'], keep='first')\n",
    "ned_asqtl = ned_asqtl.groupby([\"Gene\", \"ID\"]).agg(','.join).reset_index()\n",
    "ned_asqtl = ned_asqtl.groupby([\"ID\"]).agg(','.join).reset_index()\n",
    "\n",
    "ned_asqtl.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of intersecting SNPs: 333\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Source</th>\n",
       "      <th>ID</th>\n",
       "      <th>ESN</th>\n",
       "      <th>GWD</th>\n",
       "      <th>LWK</th>\n",
       "      <th>YRI</th>\n",
       "      <th>MSL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>21658140</td>\n",
       "      <td>both</td>\n",
       "      <td>rs34693054</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ECE1</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>22488940</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs2807362</td>\n",
       "      <td>0.005</td>\n",
       "      <td>0.009</td>\n",
       "      <td>0.040</td>\n",
       "      <td>0.009</td>\n",
       "      <td>0.000</td>\n",
       "      <td>CDC42</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>22502829</td>\n",
       "      <td>both</td>\n",
       "      <td>rs2744703</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.005</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>CDC42</td>\n",
       "      <td>NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>33229633</td>\n",
       "      <td>simonti_only</td>\n",
       "      <td>rs12022304</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.031</td>\n",
       "      <td>0.030</td>\n",
       "      <td>0.028</td>\n",
       "      <td>0.018</td>\n",
       "      <td>RBBP4,YARS</td>\n",
       "      <td>NI,NI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>33443064</td>\n",
       "      <td>both</td>\n",
       "      <td>rs735402</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.027</td>\n",
       "      <td>0.040</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.029</td>\n",
       "      <td>AK2</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome  Position        Source          ID    ESN    GWD    LWK    YRI  \\\n",
       "0           1  21658140          both  rs34693054  0.000  0.000  0.000  0.000   \n",
       "1           1  22488940  simonti_only   rs2807362  0.005  0.009  0.040  0.009   \n",
       "2           1  22502829          both   rs2744703  0.000  0.000  0.005  0.000   \n",
       "3           1  33229633  simonti_only  rs12022304  0.025  0.031  0.030  0.028   \n",
       "4           1  33443064          both    rs735402  0.000  0.027  0.040  0.000   \n",
       "\n",
       "     MSL        Gene    Set  \n",
       "0  0.000        ECE1      S  \n",
       "1  0.000       CDC42      L  \n",
       "2  0.000       CDC42     NI  \n",
       "3  0.018  RBBP4,YARS  NI,NI  \n",
       "4  0.029         AK2      L  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Intersecting SNPs in Neanderthal & Nedelec asQTL datasets\n",
    "ned_asqtl_df = allpop_df.merge(ned_asqtl, how='inner', on=['ID'])\n",
    "print(\"Number of intersecting SNPs: \" + str(len(ned_asqtl_df)))\n",
    "ned_asqtl_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1mExpression context of intersecting SNPs\u001b[0m\n",
      "NI    147\n",
      "S     135\n",
      "L     121\n",
      "dtype: int64\n",
      "\u001b[1mAssociated genes\u001b[0m\n",
      "RAD1        3\n",
      "FCGR2A      3\n",
      "DDX60L      3\n",
      "OAT         3\n",
      "SAMSN1      3\n",
      "           ..\n",
      "TGFBRAP1    1\n",
      "BRD9        1\n",
      "LHPP        1\n",
      "AHNAK       1\n",
      "PSEN1       1\n",
      "Length: 309, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Expression context of intersecting SNPs\n",
    "set = ned_asqtl_df['Set'].str.split(',', expand=True).stack().value_counts()\n",
    "print('\\033[1mExpression context of intersecting SNPs\\033[0m', set, sep='\\n')\n",
    "\n",
    "genes = ned_asqtl_df['Gene'].str.split(',', expand=True).stack().value_counts()\n",
    "print(\"\\033[1mAssociated genes\\033[0m\", genes, sep='\\n')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
