# Extreme response code

### Requirements:
Python 3 and modules:
* pandas 1.0.3
* seaborn 0.10.0
* Matplotlib 3.1.1
* SciPy 1.4.1
* plinkio 0.9.7
* networkx 2.4

R and packages:
* XGR
* RCircos

---

### Directory

The extreme response code is split into the following folders:  

* [alasoo](https://github.com/kshiyao/neanderthal_introgression/tree/master/extreme_response/alasoo)  
This folder contains the code for the recomputed Alasoo *et al.* (2018) dataset.  

* [fairfax](https://github.com/kshiyao/neanderthal_introgression/tree/master/extreme_response/fairfax)  
This folder contains the code for the Fairfax *et al.* (2014) dataset.  

* [nedelec](https://github.com/kshiyao/neanderthal_introgression/tree/master/extreme_response/nedelec)  
This folder contains the code for the Nedelec *et al.* (2016) dataset.  

* [quach](https://github.com/kshiyao/neanderthal_introgression/tree/master/extreme_response/quach)  
This folder contains the code for the recomputed Quach *et al.* (2016) dataset.  

---

### Datasets
Neanderthal-introgressed SNPs from:
1. Dannemann M, Prufer K & Kelso J. Functional implications of Neandertal introgression in modern humans. *Genome Biol* 2017 **18**:61.
2. Simonti CN *et al.* The phenotypic legacy of admixture between modern humans and Neandertals. *Science* 2016 **351**:737-41.  
&nbsp;

MAFs from:
* [1000 genomes](https://www.internationalgenome.org/data/)  
&nbsp;

Genotyped SNPs and monocyte, gene-expression data from:
* Fairfax BP *et al.* Innate immune activity conditions the effect of regulatory variants upon monocyte gene expression. *Science* 2014 **343**:1246949.  
&nbsp;

Macrophage gene-expression data from:
* Nedelec Y *et al.* Genetic Ancestry and Natural Selection Drive Population Differences in Immune Responses to Pathogens. *Cell* 2016 **167**:657-669e21.  
&nbsp;

Recomputed eQTLs from:
* [EMBL-EBI eQTL Catalogue](https://www.ebi.ac.uk/eqtl/Data_access/)  
